Ryujinx.HLE/HOS/Services/SurfaceFlinger/BufferQueueProducer.cs

```diff
@@ -0,0 +1,741 @@
+
+                listener = Core.ConsumerListener;
+            }
+
+            listener?.OnBuffersReleased();
+
+            return Status.Success;
+        }
+
+        public override Status DequeueBuffer(out int slot, out AndroidFence fence, bool async, uint width, uint height, PixelFormat format, uint usage)
+        {
+            if ((width == 0 && height != 0) || (height == 0 && width != 0))
+            {
+                slot  = BufferSlotArray.InvalidBufferSlot;
+                fence = AndroidFence.NoFence;
+
+                return Status.BadValue;
+            }
+
+            Status returnFlags = Status.Success;
+
+            bool attachedByConsumer = false;
+
+            lock (Core.Lock)
+            {
+                if (format == PixelFormat.Unknown)
+                {
+                    format = Core.DefaultBufferFormat;
+                }
+
+                usage |= Core.ConsumerUsageBits;
+
+                Status status = WaitForFreeSlotThenRelock(async, out slot, out returnFlags);
+
+                if (status != Status.Success)
+                {
+                    slot  = BufferSlotArray.InvalidBufferSlot;
+                    fence = AndroidFence.NoFence;
+
+                    return status;
+                }
+
+                if (slot == BufferSlotArray.InvalidBufferSlot)
+                {
+                    slot  = BufferSlotArray.InvalidBufferSlot;
+                    fence = AndroidFence.NoFence;
+
+                    Logger.PrintError(LogClass.SurfaceFlinger, "No available buffer slots");
+                    return Status.Busy;
+                }
+
+                attachedByConsumer = Core.Slots[slot].AttachedByConsumer;
+
+                if (width == 0 || height == 0)
+                {
+                    width  = (uint)Core.DefaultWidth;
+                    height = (uint)Core.DefaultHeight;
+                }
+
+                Core.Slots[slot].BufferState = BufferState.Dequeued;
+
+                GraphicBuffer graphicBuffer = Core.Slots[slot].GraphicBuffer.Object;
+
+                if (Core.Slots[slot].GraphicBuffer.IsNull
+                    || graphicBuffer.Width != width 
+                    || graphicBuffer.Height != height 
+                    || graphicBuffer.Format != format
+                    || (graphicBuffer.Usage & usage) != usage)
+                {
+                    if (Core.Slots[slot].GraphicBuffer.IsNull)
+                    {
+                        slot  = BufferSlotArray.InvalidBufferSlot;
+                        fence = AndroidFence.NoFence;
+
+                        return Status.NoMemory;
+                    }
+                    else
+                    {
+                        string formattedError = $"Preallocated buffer mismatch - slot {slot}\n" +
+                                               $"available: Width = {graphicBuffer.Width} Height = {graphicBuffer.Height} Format = {graphicBuffer.Format} Usage = {graphicBuffer.Usage:x} " +
```

Comment:
```suggestion
                                                $"available: Width = {graphicBuffer.Width} Height = {graphicBuffer.Height} Format = {graphicBuffer.Format} Usage = {graphicBuffer.Usage:x} " +
```
And same for the next line